### PR TITLE
RavenDB upgrade guide: Migrate saga data if using AllowStaleSagaReads

### PR DESCRIPTION
**Pull request by DavidBoike**
Connects to https://github.com/Particular/PlatformDevelopment/issues/1443

### CODE REVIEW

File: persistence/upgrades/ravendb-3to4.md [hunk 1]
```diff
@@ -37,6 +37,21 @@ NServiceBus now uses the asynchronous RavenDB API for all operations. If sharing
 Previously the API exposed an [`IDocumentSession`](https://ravendb.net/docs/search/latest/csharp?searchTerm=IDocumentSession), but now exposes [`IAsyncDocumentSession`](https://ravendb.net/docs/search/latest/csharp?searchTerm=IAsyncDocumentSession) instead, which contains the same operations but using a Task-based API.
 
 
+## Migrate saga data if using AllowStaleSagaReads
+
+Due to changes in how NServiceBus Version 6 handles saga correlation properties, solutions that previously used the `AllowStaleSagaReads()` option in RavenDB Persistence will not work properly and need to be migrated during upgrade.
```
Comment (SimonCropp) [inline persistence/upgrades/ravendb-3to4.md]: "changes in how NServiceBus Version 6 handles saga correlation properties" assuming is should be possible to add a more info link here?

File: persistence/upgrades/ravendb-3to4.md [hunk 2]
```diff
@@ -37,6 +37,21 @@ NServiceBus now uses the asynchronous RavenDB API for all operations. If sharing
 Previously the API exposed an [`IDocumentSession`](https://ravendb.net/docs/search/latest/csharp?searchTerm=IDocumentSession), but now exposes [`IAsyncDocumentSession`](https://ravendb.net/docs/search/latest/csharp?searchTerm=IAsyncDocumentSession) instead, which contains the same operations but using a Task-based API.
 
 
+## Migrate saga data if using AllowStaleSagaReads
```
Comment (SimonCropp) [inline persistence/upgrades/ravendb-3to4.md]: given the risk here. should this item be close to the top of the page?

File: persistence/upgrades/ravendb-3to4.md [hunk 3]
```diff
@@ -37,6 +37,21 @@ NServiceBus now uses the asynchronous RavenDB API for all operations. If sharing
...
+
+Due to changes in how NServiceBus Version 6 handles saga correlation properties, solutions that previously used the `AllowStaleSagaReads()` option in RavenDB Persistence will not work properly and need to be migrated during upgrade.
+
+DANGER: Failure to migrate saga data when using the `AllowStaleSagaReads()` option will result in NServiceBus being unable to locate saga data, and then creating duplicate saga data erroneously.
+
+Normally, RavenDB persistence will use a _pointer document_ to efficiently load the saga data document by the correlation property in an atomic and consistent manner, without using RavenDB indexes that are (by design) not updated atomically with the document store.
```
Comment (SimonCropp) [inline persistence/upgrades/ravendb-3to4.md]: "use a _pointer document_ to efficiently load" sounds weird? perhaps something along the line of "to enable efficient loading.."?

File: persistence/upgrades/ravendb-3to4.md [hunk 4]
```diff
@@ -37,6 +37,21 @@ NServiceBus now uses the asynchronous RavenDB API for all operations. If sharing
...
+
+When using the `AllowStaleSagaReads()` option in previous versions, which was sometimes used to support correlating saga data on multiple properties, pointer documents were not used and RavenDB indexes (which might be stale) were used instead.
+
+Starting in NServiceBus Version 6, only one correlation id is supported, and the `AllowStaleSagaReads()` option is deprecated. However, saga data stored using this option in previous versions will not have the unique identity pointer document, and will not be able to load. The result is that NServiceBus will not be able to find the saga data document. If the message handler is implemented in the saga as `IHandleMessages<T>` then the message will be incorrectly discarded as belonging to a saga that has already completed. If the message handler is implemented in the saga as `IAmStartedByMessages<T>` then a new saga data will (incorrectly) be created, leading to duplicate saga data documents and incorrect business execution.
+
+When upgrading, if the `AllowStaleSagaReads` option is in use, contact [support@particular.net](mailto:support@particular.net) for assistance in identifying the scope of the problem and migration of data.
```
Comment (SimonCropp) [inline persistence/upgrades/ravendb-3to4.md]: is `AllowStaleSagaReads` marked with some kind of obsolete warning? should it be?

Comment (DavidBoike) [reply persistence/upgrades/ravendb-3to4.md]: @SimonCropp Yeah, `AllowStaleSagaReads` is marked with an obsolete already - we're considering whether to make the message more forceful.

**Pull request merged**